pulsar-functions/runtime/src/main/java/org/apache/pulsar/functions/runtime/KubernetesRuntimeFactory.java

```diff
@@ -340,8 +353,21 @@ void validateMinResourcesRequired(Function.FunctionDetails functionDetails) {
         }
     }
 
+    KubernetesManifestCustomizer buildManifestCustomizer(Function.FunctionDetails funcDetails) {
+
+        KubernetesManifestCustomizer manifestCustomizer;
+        if (!isEmpty(manifestCustomizerClassName)) {
+            manifestCustomizer = (KubernetesManifestCustomizer) Reflections.createInstance(manifestCustomizerClassName, ClassLoader.getSystemClassLoader());
+        } else {
+            manifestCustomizer = new DefaultKubernetesManifestCustomizer();
```

Comment:
I would suggest that the default to be a no-op implementation.  Allowing users the change the certain kubernetes settings when submitting a function can be a security risk.